.github/workflows/validate.yml

```
@@ -240,20 +245,41 @@ jobs:
       - name: Run benchmarks
         run: |-
           set -euo pipefail
-          go test -bench=. -timeout=1h -run=^$ . | tee ${{ runner.temp }}/benchmarks.txt
+          go test -bench=. -timeout=1h -run=^$ . | tee ${{ runner.temp }}/benchmarks-${{ matrix.run_number }}.txt
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Upload benchmark report (raw)
         uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4
         with:
           if-no-files-found: error
-          name: Benchmark Report
-          path: ${{ runner.temp }}/benchmarks.txt
-      - name: Format Report
+          name: Benchmark Report (run ${{ matrix.run_number }})
+          path: ${{ runner.temp }}/benchmarks-${{ matrix.run_number }}.txt
+
+  benchmark-report:
+    needs: benchmark
+    runs-on: ubuntu-latest
+    name: Benchmark Report
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+      - name: Setup go
+        id: setup-go
+        uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5
+        with:
+          go-version: stable
+          cache-dependency-path: "**/go.mod"
+      - name: Download benchmark reports (raw)
+        uses: actions/download-artifact@cc203385981b70ca67e1cc392babf9cc229d5806 # v4
+        with:
+          pattern: Benchmark Report (run *)
```

doing globbing with with spaces in them sure is wierd. It took me sometime to understand
```suggestion
          name: benchmarks-${{ matrix.run_number }}.txt
          path: ${{ runner.temp }}/benchmarks-${{ matrix.run_number }}.txt

  benchmark-report:
    needs: benchmark
    runs-on: ubuntu-latest
    name: Benchmark Report
    steps:
      - name: Checkout
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
      - name: Setup go
        id: setup-go
        uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5
        with:
          go-version: stable
          cache-dependency-path: "**/go.mod"
      - name: Download benchmark reports (raw)
        uses: actions/download-artifact@cc203385981b70ca67e1cc392babf9cc229d5806 # v4
        with:
          pattern: benchmarks-*.txt
```